### PR TITLE
win, unix: extract common fields out of the loop fields.

### PR DESCRIPTION
**Pull request by txdv**
only -11 lines though in total.

### CODE REVIEW

File: include/uv.h [hunk 1]
```diff
@@ -1461,6 +1461,25 @@ struct uv_loop_s {
   /* Internal flag to signal loop stop. */
   unsigned int stop_flag;
   void* reserved[4];
+  /* private fields that appear in both implementations */
+  /* Lists of active loop (prepare / check / idle) watchers */
+  void* prepare_handles[2];
+  void* check_handles[2];
+  void* async_handles[2];
+  /* Timers */
+  void* idle_handles[2];
```
Comment (saghul) [inline include/uv.h]: move this before prepare, it doesn't belong under the Timers title

File: include/uv.h [hunk 2]
```diff
@@ -1461,6 +1461,25 @@ struct uv_loop_s {
...
+    void* min;
+    unsigned int nelts;
+  } timer_heap;
+  /* Threadpool */
+  void* wq[2];
+  uv_mutex_t wq_mutex;
+  uv_async_t wq_async;
+  /* The current time according to the event loop. in msecs. */
+  uint64_t time;
+  uint64_t timer_counter;
```
Comment (saghul) [inline include/uv.h]: put the counter together with the timer stuff

File: include/uv.h (hunk 1 above)
Comment (txdv) [inline include/uv.h]: yeah, made a mistake

Comment (txdv) [inline include/uv.h]: Em, do you want me to udpate the comment too then? Is it (async/ prepare/ check/ idle) watchers?

File: include/uv.h [hunk 3]
```diff
@@ -1461,6 +1480,25 @@ struct uv_loop_s {
...
   void* reserved[4];
+  /* private fields that appear in both implementations */
+  /* Lists of active loop (async / prepare / check / idle) watchers */
+  void* async_handles[2];
+  void* prepare_handles[2];
+  void* check_handles[2];
+  void* idle_handles[2];
+  /* The current time according to the event loop. in msecs. */
+  uint64_t time;
+  uint64_t timer_counter;
```
Comment (saghul) [inline include/uv.h]: move this to after the timer heap

**Comment by saghul** [comment]
Small nit, otherwise LGTM.

**Comment by txdv** [comment]
-30 yeah... squash into one big one?

File: include/uv.h [hunk 4]
```diff
@@ -409,6 +409,37 @@ struct uv_shutdown_s {
   } u;                                                                        \
   UV_HANDLE_PRIVATE_FIELDS                                                    \
 
+#define UV_TIMER_PRIVATE_FIELDS                                               \
```
Comment (saghul) [inline include/uv.h]: come to think of it, we no longer need these private markers if we directly put the fields in the strctures directly.  though this way may look a bit nicer.  can you check if there are private marked fields that should be extracted?

Comment (txdv) [inline include/uv.h]: > can you check if there are private marked fields that should be extracted?

I do not understand this request. Can you elaborate it a bit differently?

Comment (saghul) [inline include/uv.h]: On some handles we have some fields directly in their struct definition which are marked as private with a comment.  Check if there are repetitions please.

Also, ignore the first part of my message, I think it looks better with the define, for now.

**Comment by txdv** [comment]
I have found some double definitions in

UV_ASYNC_PRIVATE_FIELDS
UV_FS_PRIVATE_FIELDS
UV_FS_EVENT_PRIVATE_FIELDS

as in, there are fields which are both present.

UV_FS_PRIVATE_FIELDS has this gem: https://github.com/txdv/libuv/blob/v1.x/include/uv-win.h#L600

Should I pull out the common fields and put them just in the struct definition?

**Comment by txdv** [comment]
I pulled out common variables of all the other structs.

I noticed that sometimes the callbacks in the structs are called just cb, some they are called shutdown_cb(<action_name>_cb). I think <action_name>_cb is better, should I rename all of them in a similar style?

**Comment by saghul** [comment]
Let's leave that for another PR.  This one is about the common fields, then we can consolidate naming.

**Comment by txdv** [comment]
Ok, so the only one left is UV_FS_PRIVATE_FIELDS.

That one has a super old [TODO](https://github.com/libuv/libuv/commit/514265ec6ec79365f8e691b6a02ed9bb6ae128cd) on windows and no union on unix (I think it is possible to put most of the fields in unions, depending on what operation is being done, to save some space).

I would like @piscisaureus to make a comment on that TODO

**Comment by txdv** [comment]
No comments, nothing?

**Comment by saghul** [comment]
Sorry, I have some busy time ahe.  I'll come back to this as soon as I can.  Thanks for your work!

**Comment by txdv** [comment]
@saghul can you run the integration tests?

**Comment by txdv** [comment]
With https://github.com/libuv/libuv/pull/1255 I realized that https://github.com/txdv/libuv/blob/slimmer-fs-struct/include/uv.h#L416-L417 timer and repeat should be public fields, should I put them at the top and comment them like that?

**Pull request closed without merging**